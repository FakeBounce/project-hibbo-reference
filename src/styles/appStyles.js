/* eslint global-require: 0 */
import { Dimensions } from 'react-native';

import { iconVideos } from 'utils/storedData';

const { width } = Dimensions.get('window');

export default {
  prefix: {
    fixedIcon: 'icon_opac/',
  },
  event: {
    none: 0,
    gateau: 1,
    sapin: 2,
    oeuf: 3,
    souris: 4,
    bonbon: 11,
    manette: 12,
    moto: 13,
    voiture: 14,
    cadeau_vert: 15,
    cadeau_bleu: 16,
    courses: 21,
    rangement: 22,
    crayon: 23,
    marteau: 24,
    lapin: 31,
    chat: 32,
    chien: 33,
    heureux: 41,
    betise: 42,
    system: {
      reset: 0xffff,
      setLocal: 0xfff0,
      setVolume: 0xfff1,
      setObjectif: 0xfff2,
    },
  },
  icons: iconVideos,
  maxTextSize: 70,
  colors: {
    white: '#FFF',
    black: '#000',
    darkGrey: '#303030',
    darkBlue: '#06263D',
    darkBlueModal: '#06263D',
    grey: '#979797',
    greyOpac: 'rgba(151, 151, 151, 0.18)',
    whiteOpac: 'rgba(255, 255, 255, 0.30)',
    green: '#16EBC5',
    fb: '#4100C0',
    google: '#D5492F',
    primary: '#413D3D',
    textGrey: '#A7A7A7',
    buttonGrey: '#EFEFEF',
    lightGrey: '#9B9B9B',
    lightGreyBullet: '#E5E3DD',
    blueGreen: '#16EBC5',
    borderGrey: 'rgba(74, 74, 74, 0.23)',
    blackModal: 'rgba(0, 0, 0, 0.5)',
    toggleButtonGrey: '#4A4A4A',
    yellow: '#FFDF2D',
    moneyPotText: '#FFFE82',
    buttonYellow: '#FFFE82',
    cardBlue: '#697A99',
    gold: '#FEE9BF',
    red: '#FC5151',
    error: '#FF6969',
    textLink: '#687A99',
    messageGrey: '#DDE6EC',
    blue: '#0084ff',
    balanceGreen: '#3CDD80',
    pelicanColor: '#A685FF',
    deepBlue: '#028DAF',
    flatPink: '#FF6682',
    flatOrange: '#FFB434',
    flatPurple: '#82438F',
    white2pac: 'rgba(255,255,255,0.2)',
    white1pac: 'rgba(255,255,255,0.1)',
    placeholderGrey: 'rgba(229, 227, 221, 0.5)',
    greenStrong: '#50E3C2',
    orangeNotStrong: '#F5A623',
    inputLabel: '#C3C3C3',
    inputGrey: '#E8E8E8',
  },
  transaction: {
    minimunForAnimation: 1000,
  },
  time: {
    moneyTransaction: 2500,
  },
  textSizes: {
    xxxsmall: 10,
    xxsmall: 13,
    xsmall: 16,
    small: 18,
    regular: 22,
    normal: 24,
    xnormal: 26,
    big: 36,
    xbig: 50,
    xxbig: 58,
  },
  size: {
    cardMargin: 10,
    maxBubbleWidth: 190,
  },
  cardStyle: {
    shadowOpacity: 0,
    shadowColor: 'transparent',
    borderRadius: 10,
  },
  fonts: {
    circularBook: 'Circular-Std-Book',
    circularBookItalic: 'Circular-Std-Book-Italic',
    circularMedium: 'Circular-Std-Medium',
    circularMediumItalic: 'Circular-Std-Medium-Italic',
    circularBold: 'Circular-Std-Bold',
    circularBoldItalic: 'Circular-Std-Bold-Italic',
    circularBlackItalic: 'Circular-Std-Black-Italic',
    pixel: 'LCDDot-TR',
    chewy: 'Chewy',
    peepo: 'Peepo',
  },
  get colorList() {
    return [
      this.colors.deepBlue,
      this.colors.flatPink,
      this.colors.flatOrange,
      this.colors.flatPurple,
    ];
  },
  colorListSuffix: ['_bleu', '_rose', '_orange', '_violet'],
  swiperRange: [0, width / 2, width, width * 1.5, width * 2],
};
