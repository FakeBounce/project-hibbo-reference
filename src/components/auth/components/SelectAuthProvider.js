import React, { PureComponent } from 'react';
import { Animated } from 'react-native';
import { string, func, shape, instanceOf } from 'prop-types';
import EStyleSheet from 'react-native-extended-stylesheet';
import { bind } from 'decko';

import { getTranslations } from 'utils/i18n';

import { StyledContainerBasic } from 'styledComponents/containers';

import fb from 'assets/auth/icons/fb.png';
import google from 'assets/auth/icons/google.png';
import mail from 'assets/auth/icons/mail.png';

import AuthProviderErrors from './AuthProviderErrors';
import ProviderLinks from './ProviderLinks';

import * as Styles from '../styles';

const styles = EStyleSheet.create({
  fbText: {
    color: '$colors.fb',
    fontSize: '$textSizes.small',
    fontFamily: '$fonts.circularBook',
  },
  fb: {
    width: 9,
    height: 17,
    marginRight: 15,
  },
  googleText: {
    color: '$colors.google',
    fontSize: '$textSizes.small',
    fontFamily: '$fonts.circularBook',
  },
  google: {
    width: 19,
    height: 19,
    marginRight: 15,
  },
  mailText: {
    color: '$colors.green',
    fontSize: '$textSizes.small',
    fontFamily: '$fonts.circularBook',
  },
  mail: {
    width: 21,
    height: 15,
    marginRight: 15,
  },
});

class SelectAuthProvider extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      fbPosition: {
        x: 0,
        y: 0,
      },
      googlePosition: {
        x: 0,
        y: 0,
      },
    };
  }

  @bind
  handlePosition(name, position) {
    this.setState(state => {
      return {
        ...state,
        [name]: position,
      };
    });
  }

  render() {
    const {
      authType,
      goToNextStep,
      registerWithGoogle,
      loginWithGoogle,
      registerWithFb,
      loginWithFb,
      styling,
      fbError,
      googleError,
      selectAuthProvider,
      ...rest
    } = this.props;
    const { fbPosition, googlePosition } = this.state;

    return (
      <Styles.AContainerAuthProv style={styling}>
        <AuthProviderErrors
          googleError={googleError}
          googlePosition={googlePosition}
          fbError={fbError}
          fbPosition={fbPosition}
        />

        <StyledContainerBasic>
          <ProviderLinks
            name="facebook"
            label={getTranslations('auth.provider.fb')}
            labelStyle={styles.fbText}
            imageSource={fb}
            imageStyle={styles.fb}
            elementPosition="fbPosition"
            handlePosition={this.handlePosition}
            onPress={() => {
              selectAuthProvider('facebook');
              if (authType === 'register') {
                registerWithFb();
              } else {
                loginWithFb();
              }
            }}
            bordered
            {...rest}
          />

          <ProviderLinks
            name="google"
            label={getTranslations('auth.provider.google')}
            labelStyle={styles.googleText}
            imageSource={google}
            imageStyle={styles.google}
            elementPosition="googlePosition"
            handlePosition={this.handlePosition}
            onPress={() => {
              selectAuthProvider('google');
              if (authType === 'register') {
                registerWithGoogle();
              } else {
                loginWithGoogle();
              }
            }}
            bordered
            {...rest}
          />

          <ProviderLinks
            name="email"
            label={getTranslations('auth.provider.mail')}
            labelStyle={styles.mailText}
            imageSource={mail}
            imageStyle={styles.mail}
            onPress={() => {
              goToNextStep(2);
            }}
            bordered={false}
            {...rest}
          />
        </StyledContainerBasic>
      </Styles.AContainerAuthProv>
    );
  }
}

SelectAuthProvider.propTypes = {
  authType: string.isRequired,
  goToNextStep: func.isRequired,
  registerWithGoogle: func.isRequired,
  loginWithGoogle: func.isRequired,
  registerWithFb: func.isRequired,
  loginWithFb: func.isRequired,
  fbError: string.isRequired,
  googleError: string.isRequired,
  styling: shape({
    height: instanceOf(Animated.Value).isRequired,
    opacity: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  selectAuthProvider: func.isRequired,
};

export default SelectAuthProvider;
