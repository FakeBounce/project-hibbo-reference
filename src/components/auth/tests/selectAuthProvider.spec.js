import { Animated } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EStyleSheet from 'react-native-extended-stylesheet';

import appStyles from 'styles/appStyles';

import SelectAuthProvider from '../components/SelectAuthProvider';

EStyleSheet.build(appStyles);

describe('SelectAuthProvider', () => {
  const styling = {
    height: new Animated.Value(300),
    opacity: new Animated.Value(1),
  };

  it('should render the component', () => {
    const fn = jest.fn();
    const component = renderer
      .create(
        <SelectAuthProvider
          authType=""
          goToNextStep={fn}
          registerWithGoogle={fn}
          loginWithGoogle={fn}
          registerWithFb={fn}
          loginWithFb={fn}
          fbError=""
          googleError=""
          styling={styling}
        />,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should handle error from fb', () => {
    const fn = jest.fn();
    const component = renderer
      .create(
        <SelectAuthProvider
          authType=""
          goToNextStep={fn}
          registerWithGoogle={fn}
          loginWithGoogle={fn}
          registerWithFb={fn}
          loginWithFb={fn}
          fbError="Error"
          googleError=""
          styling={styling}
        />,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should handle error from google', () => {
    const fn = jest.fn();
    const component = renderer
      .create(
        <SelectAuthProvider
          authType=""
          goToNextStep={fn}
          registerWithGoogle={fn}
          loginWithGoogle={fn}
          registerWithFb={fn}
          loginWithFb={fn}
          fbError=""
          googleError="Error"
          styling={styling}
        />,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
