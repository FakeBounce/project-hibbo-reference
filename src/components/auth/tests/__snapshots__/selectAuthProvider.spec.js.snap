// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAuthProvider should handle error from fb 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "#FFF",
      "bottom": 0,
      "flexDirection": "row",
      "height": 300,
      "opacity": 1,
      "position": "absolute",
      "width": 750,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "elevation": 10,
          "position": "absolute",
          "zIndex": 10,
        },
        Object {
          "left": 50,
          "top": -55,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 100,
            "height": 75,
            "justifyContent": "center",
            "width": 275,
          },
          Object {
            "backgroundColor": "#FF7171",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 10,
            }
          }
        >
          <Image
            source={1}
            style={undefined}
          />
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#EFEFEF",
                "fontFamily": "Circular-Std-Book",
              }
            }
          >
            Error
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomWidth": 50,
              "borderLeftColor": "transparent",
              "borderLeftWidth": 20,
              "borderRightColor": "transparent",
              "borderRightWidth": 20,
              "borderStyle": "solid",
              "elevation": -1,
              "height": 0,
              "position": "absolute",
              "width": 0,
              "zIndex": -1,
            },
            Object {
              "bottom": -20,
              "left": 25,
              "transform": Array [
                Object {
                  "rotate": "160deg",
                },
              ],
            },
            Object {
              "borderBottomColor": "#FF7171",
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 17,
            "marginRight": 15,
            "width": 9,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4100C0",
            "fontSize": 19,
          }
        }
      >
        Facebook connect
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 19,
            "marginRight": 15,
            "width": 19,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#D5492F",
            "fontSize": 19,
          }
        }
      >
        Google connect
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 15,
            "marginRight": 15,
            "width": 21,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16EBC5",
            "fontSize": 19,
          }
        }
      >
        Email connect
      </Text>
    </View>
  </View>
</View>
`;

exports[`SelectAuthProvider should handle error from google 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "#FFF",
      "bottom": 0,
      "flexDirection": "row",
      "height": 300,
      "opacity": 1,
      "position": "absolute",
      "width": 750,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "elevation": 10,
          "position": "absolute",
          "zIndex": 10,
        },
        Object {
          "left": 50,
          "top": -55,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 100,
            "height": 75,
            "justifyContent": "center",
            "width": 275,
          },
          Object {
            "backgroundColor": "#FF7171",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 10,
            }
          }
        >
          <Image
            source={1}
            style={undefined}
          />
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#EFEFEF",
                "fontFamily": "Circular-Std-Book",
              }
            }
          >
            Error
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomWidth": 50,
              "borderLeftColor": "transparent",
              "borderLeftWidth": 20,
              "borderRightColor": "transparent",
              "borderRightWidth": 20,
              "borderStyle": "solid",
              "elevation": -1,
              "height": 0,
              "position": "absolute",
              "width": 0,
              "zIndex": -1,
            },
            Object {
              "bottom": -20,
              "left": 25,
              "transform": Array [
                Object {
                  "rotate": "160deg",
                },
              ],
            },
            Object {
              "borderBottomColor": "#FF7171",
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 17,
            "marginRight": 15,
            "width": 9,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4100C0",
            "fontSize": 19,
          }
        }
      >
        Facebook connect
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 19,
            "marginRight": 15,
            "width": 19,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#D5492F",
            "fontSize": 19,
          }
        }
      >
        Google connect
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 15,
            "marginRight": 15,
            "width": 21,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16EBC5",
            "fontSize": 19,
          }
        }
      >
        Email connect
      </Text>
    </View>
  </View>
</View>
`;

exports[`SelectAuthProvider should render the component 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "#FFF",
      "bottom": 0,
      "flexDirection": "row",
      "height": 300,
      "opacity": 1,
      "position": "absolute",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 17,
            "marginRight": 15,
            "width": 9,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4100C0",
            "fontSize": 19,
          }
        }
      >
        Facebook connect
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 19,
            "marginRight": 15,
            "width": 19,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#D5492F",
            "fontSize": 19,
          }
        }
      >
        Google connect
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": 750,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 15,
            "marginRight": 15,
            "width": 21,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#16EBC5",
            "fontSize": 19,
          }
        }
      >
        Email connect
      </Text>
    </View>
  </View>
</View>
`;
